opennms-base-assembly/src/main/filtered/bin/find-java.sh

```diff
@@ -50,11 +50,11 @@ get_java_version_string() {
 get_real_path() {
 	file_to_find="$1"
 	readlink="$(command -v readlink 2>/dev/null)"
-	if [ -n "$readlink" ] && [ -n "$file_to_find" ] && [ -e "$file_to_find" ]; then
-		"$readlink" "$file_to_find" || echo "$file_to_find"
-	else
-		echo "$file_to_find"
+	if [ -n "$readlink" ] && [ -L "$file_to_find" ]; then
+		get_real_path "$("$readlink" "$file_to_find" || :)"
```

Comment:
Is the no-op necessary here? We aren't doing anything with the return code.
```suggestion
		get_real_path "$("$readlink" "$file_to_find")"
```

Reply:
true, previously I was detecting if it returns nothing, but now since it's guarded by `-L`, failure should actually shortcut out as an error